pkg/utils/utils.go

```diff
@@ -733,6 +733,49 @@ func GetSecretNamespaceAndName(kubeClient kubernetes.Interface, veleroNs string,
 	return namespace, name
 }
 
+func GetUploadCRRetryMaximumFromConfig(kubeClient kubernetes.Interface, veleroNs string, configName string) int {
+	veleroPuginConfigMap, err := kubeClient.CoreV1().ConfigMaps(veleroNs).Get(context.TODO(), configName, metav1.GetOptions{})
+	if err != nil {
+		fmt.Printf("failed to retrieve %s configuration, err: %+v.\n", configName, err)
+		return constants.DefaultUploadCRRetryMaximum
+	}
+	configMap := veleroPuginConfigMap.Data
+	if maxRetryCountStr, ok := configMap[constants.UploadCRRetryMaximumKey]; ok {
+		maxRetryCount, err := strconv.Atoi(maxRetryCountStr)
+		if err != nil {
+			fmt.Printf("UploadCRRretryMaximum %s is invalid, err: %+v.\n", maxRetryCountStr, err)
+			maxRetryCount = constants.DefaultUploadCRRetryMaximum
+		}
+		return maxRetryCount
+	}
+	return constants.DefaultUploadCRRetryMaximum
+}
+
+func GetUploadCRRetryMaximum(config *rest.Config) int {
+	var err error
+	if config == nil {
+		config, err = GetKubeClientConfig()
+		if err != nil {
+			return constants.DefaultUploadCRRetryMaximum
+		}
+	}
+
+	clientset, err := GetKubeClientSet(config)
+	if err != nil {
+		return constants.DefaultUploadCRRetryMaximum
```

Comment:
log the error, someone debugging should be aware that the retry maximum was not honored.